draft-ietf-tls-wkech.txt

```diff
@@ -131,157 +139,155 @@ Internet-Draft           Well-Known URI for ECH             October 2022
+   value to have already been published in the DNS.  [[I guess that
+   calls for the zone factory to know of a "safe" URL on $ORIGIN to try,
+   or maybe it could use HTTP HEAD?  Figuring that out is TBD.  The ZF
+   could also try a GREASEd ECH and see if the retry-configs it gets
+   back is one of the ECHConfig values in the ECHConfigList.]]
 
    A careful zone factory could explode the ECHConfigList value
    presented into "singleton" values with one public key in each and
-   test each for each port claimed.
+   test each for each endpoint.
 
-   The zone factory SHOULD publish all the ECHConfigList values that are
-   presented in the JSON file, and that pass the check above.
+   The zone factory SHOULD publish all the endpoints that are presented
+   in the JSON file, and that pass the check above.
 
-   The zone factory SHOULD only publish ECHConfigList values that are in
-   the latest version of the JSON file.  This leaves the control of
-   "expiry" with the web server, so long as the ECHConfigList values
-   presented actually work.  [[An alternative could be to have the new
-   values just be appended to the zone, but that'd require some form of
-   "notAfter" value in the JSON file which seems unnecessary and more
-   complex.]]
+   The zone factory MUST set a DNS TTL short enough that any generated
+   records expire from DNS caches before the JSON object's HTTP cache
+   lifetime expires.  The zone factory MUST refresh the JSON object and
+   regenerate the zone before it expires each time.  This ensures that
+   ECHConfigs are not used longer than intended by the origin, while
+   permitting the zone factory to limit the TTL if desired.
 
-   The SCVB and HTTPS RR specification [I-D.ietf-dnsop-svcb-https]
-   defines how and where the ECHConfigList values for $BACKEND needs to
-   be published in the DNS.  The zone factory is assumed to be in
-   control of how ECHConfigList values are included in such RRs.
-
-   A possibly interesting (unintended) consequence of this design is
-   that once a TLS client has first gotten an ECHConfigList from the DNS
-   for $BACKEND with the ECHConfigList structure containing the
-   public_name field, the TLS client would know both $FRONT and $BACKEND
-   and so could later probe for this .well-known as an alternative to
-   doing so via DoT/DoH.  Probably not something a web browser might do,
-   but could be fun for other applications maybe.
 
-   [[The extent to which retry-configs could be used for a similar
-   purpose might be worth considering.  But the JSON stuff here may
-   still be needed if implementations (such as mine:-) tend to only
-   return one ECHConfig in retry-configs.]]
 
 
 
-Farrell, et al.          Expires April 17, 2023                 [Page 5]
+Farrell, et al.          Expires April 23, 2023                 [Page 5]
 
 Internet-Draft           Well-Known URI for ECH             October 2022
 
 
+   A possibly interesting (unintended) consequence of this design is
```

Comment:
Well, this whole paragraph is kinda casual and cutesy. Hard to see justification for unintended tho.  And not sure "fun" is appropriate.

Reply:
OK, I've rewritten this in formal terms and moved it to the Security Considerations.